lib/active_admin/views/components/active_admin_form.rb

```diff
@@ -131,12 +131,22 @@ def build(form_builder, *args, &block)
         html_options[:class] ||= "inputs"
         legend = args.shift if args.first.is_a?(::String)
         legend = html_options.delete(:name) if html_options.key?(:name)
-        legend_tag = legend ? "<legend class=\"fieldset-title\">#{ERB::Util.html_escape(legend)}</legend>" : ""
-        fieldset_attrs = html_options.map { |k, v| %Q{#{k}="#{v}"} }.join(" ")
+        legend_tag = legend ? tag.legend(legend, class: "fieldset-title") : ""
+        fieldset_attrs = tag_attributes html_options
         @opening_tag = "<fieldset #{fieldset_attrs}>#{legend_tag}<ol>"
         @closing_tag = "</ol></fieldset>"
```

Comment:
Note that it's not so trivial to use rails helpers to build the `fieldset` and `ol` since they are fragments.